src/transformers/modeling_tf_utils.py

```diff
@@ -2570,12 +2573,6 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
                         f"Error no file named {TF2_WEIGHTS_NAME} or {WEIGHTS_NAME} found in directory "
                         f"{pretrained_model_name_or_path}."
                     )
-            elif os.path.isfile(pretrained_model_name_or_path):
-                archive_file = pretrained_model_name_or_path
-                is_local = True
-            elif os.path.isfile(pretrained_model_name_or_path + ".index"):
-                archive_file = pretrained_model_name_or_path + ".index"
-                is_local = True
```

Comment:
This code shouldn't be removed, to preserve compatibility with `PreTrainedModel.from_pretrained(path_to_a_model_path, config = config)`